File: src/main/java/me/andre111/d20server/command/GetCommand.java
package me.andre111.d20server.command;

import me.andre111.d20common.model.entity.ChatEntry;
import me.andre111.d20common.model.entity.profile.Profile;
import me.andre111.d20common.scripting.Context;
import me.andre111.d20common.scripting.ScriptException;
import me.andre111.d20common.scripting.variable.Variable;
import me.andre111.d20common.scripting.variable.parser.VariableParser;
import me.andre111.d20server.service.ChatService;

public class GetCommand extends Command {
	public GetCommand(String name, String[] aliases) {
		super(name, aliases);
	}

	@Override
	public void execute(Profile profile, String arguments) {
		try {
			Variable variable = VariableParser.parseVariable(arguments);
			Object value = variable.get(new Context(profile, profile.getMap(), null));
			
			StringBuilder sb = new StringBuilder();
			sb.append(ChatService.STYLE_INFO);
			sb.append(arguments);
			sb.append(" = ");
			sb.append(value);
			
			ChatService.append(false, new ChatEntry(sb.toString(), ChatService.SYSTEM_SOURCE, false, profile.id()));
		} catch (ScriptException e) {
			ChatService.appendError(profile, "Could not get "+arguments+":", e.getMessage());
		}
	}
}

File: core/server/command/get-command.js
import { EntityManagers } from '../../common/entity/entity-managers.js';
import { Context } from '../../common/scripting/context.js';
import { parseVariable } from '../../common/scripting/variable/parser/variable-parsers.js';
import { ChatService } from '../service/chat-service.js';
import { Command } from './command.js';

export class GetCommand extends Command {
    constructor(name, aliases) {
        super(name, aliases, false);
    }

    execute(profile, args) {
        const variable = parseVariable(args);
        const value = variable.get(new Context(profile, EntityManagers.get('map').find(profile.getCurrentMap()), null));

        ChatService.appendNote(profile, ChatService.escape(args), ' = ' + ChatService.escape(''+value));
    }
}
